tests/test_models/test_nn/test_timesfm.py

```diff
@@ -0,0 +1,191 @@
+
+
+@pytest.mark.smoke
+def test_chronos_get_model(example_tsds):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch")
+    assert isinstance(model.get_model(), TimesFmTorch)
+
+
+@pytest.mark.smoke
+def test_fit(example_tsds):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch")
+    model.fit(example_tsds)
+
+
+@pytest.mark.smoke
+def test_predict(example_tsds):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch")
+    with pytest.raises(NotImplementedError, match="Method predict isn't currently implemented!"):
+        model.predict(ts=example_tsds, prediction_size=1)
+
+
+@pytest.mark.smoke
+def test_forecast_warns_big_context_size(ts_increasing_integers):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch", encoder_length=512)
+    pipeline = Pipeline(model=model, horizon=1)
+    pipeline.fit(ts_increasing_integers)
+    with pytest.warns(UserWarning, match="Actual length of a dataset is less that context size."):
+        _ = pipeline.forecast()
+
+
+@pytest.mark.parametrize("encoder_length", [32, 64, 128])
+def test_forecast(ts_increasing_integers, expected_ts_increasing_integers, encoder_length):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch", encoder_length=encoder_length)
+    pipeline = Pipeline(model=model, horizon=2)
+    pipeline.fit(ts_increasing_integers)
+    forecast = pipeline.forecast()
+    assert_frame_equal(forecast.df, expected_ts_increasing_integers.df, atol=1)
+
+
+def test_forecast_exogenous_features(ts_increasing_integers, expected_ts_increasing_integers):
+    horizon = 2
+    transforms = [
+        SegmentEncoderTransform(),
+        LagTransform(in_column="target", lags=[horizon], out_column="lag"),
+        DateFlagsTransform(day_number_in_week=True, day_number_in_month=False, is_weekend=False, out_column="flag"),
+    ]
+    model = TimesFMModel(
+        path_or_url="google/timesfm-1.0-200m-pytorch",
+        encoder_length=32,
+        static_categoricals=["segment_code"],
+        time_varying_reals=[f"lag_{horizon}"],
+        time_varying_categoricals=["flag_day_number_in_week"],
+    )
+    pipeline = Pipeline(model=model, transforms=transforms, horizon=horizon)
+    pipeline.fit(ts_increasing_integers)
+    forecast = pipeline.forecast()
+    assert_frame_equal(forecast.df.loc[:, pd.IndexSlice[:, "target"]], expected_ts_increasing_integers.df, atol=1)
+
+
+@pytest.mark.smoke
+def test_forecast_only_target_failed_int_timestamps(example_tsds_int_timestamp):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch", encoder_length=32)
+    pipeline = Pipeline(model=model, horizon=1)
+    pipeline.fit(example_tsds_int_timestamp)
+    with pytest.raises(
+        NotImplementedError,
+        match="Data with None frequency isn't currently implemented for forecasting without exogenous features.",
+    ):
+        _ = pipeline.forecast()
+
+
+@pytest.mark.smoke
```

Comment:
Do we need to set all tests as smokes?

Reply:
Tests that check equality of real forecast and expected forecast are not smoke.
 I think other tests check basic functionalities

Reply:
Probably some tests aren't smokes:
- `test_forecast_warns_big_context_size`
- `test_forecast_wrong_context_len`